client/src/components/footer/change.js

```diff
@@ -31,13 +31,26 @@ class Change extends Component {
             fontSize: 0,
             textColor: false,
             bgColor: false,
-            displayColorPalette: false
+            displayColorPalette: false,
+            nextPage: "",
+            backPage: ""
         };
         this.handleClick = this.handleClick.bind(this);
         this.handleOutsideClick = this.handleOutsideClick.bind(this);
 
     }
 
+    componentWillMount() {
+        let current_idx = this.props.body;
+        if (current_idx < 4) {
+            this.setState({nextPage: Pages[current_idx + 1]});
+        }
+        if (current_idx > 0){
+            this.setState({backPage: Pages[current_idx - 1]});
+        }
+
+    }
+
```

Comment:
Returning the next page and back page works fine.

I'm not seeing any values coming from `console.log(this.state.backPage)` and `console.log(this.state.nextPage)` on render. When hardcoding the nextPage and backPage state values, they show up on the buttons so that is not the problem. `this.setState({nextPage: Pages[current_idx + 1]});` works correctly and should in fact be properly getting values from the imported Page component.

**The problem is with `this.props.body` on line 44. Console log shows that it is undefined, and therefore tries to get an undefined value from Pages and set an undefined value onto state variables.**